Domains/3-DisciplineOther/Hydraulics/SewerHydraulicAnalysis.ecschema.xml

```diff
@@ -142,9 +148,34 @@
 		<ECProperty propertyName="ManholeIsBolted" typeName="boolean" displayLabel="Is Bolted" category="HydraulicData"/>
 	</ECEntityClass>
 
-	<ECEntityClass typeName="CatchBasin" modifier="Sealed" displayLabel="Catch Basin">
-		<BaseClass>GravityStructure</BaseClass>
-	</ECEntityClass>
+  <ECEntityClass typeName="CatchBasin" modifier="Sealed" displayLabel="Catch Basin">
+    <BaseClass>GravityStructure</BaseClass>
+    <ECProperty propertyName="InletCaptureMethod" typeName="InletCaptureMethod" displayLabel="Inlet Capture Method" description="Method used to determine the rate of flow capture by the inlet." category="HydraulicData"/>
+  </ECEntityClass>
+  
+	<ECEntityClass typeName="InletCaptureMethodAspect" modifier="Abstract" displayLabel="InletCaptureMethodAspect">
+    <BaseClass>bis:ElementUniqueAspect</BaseClass>
+  </ECEntityClass>
+  
+	<ECRelationshipClass typeName="CatchBasinOwnsInletCaptureMethodAspect" strength="embedding" modifier="Sealed">
+	  <BaseClass>bis:ElementOwnsUniqueAspect</BaseClass>
+    <Source multiplicity="(1..1)" roleLabel="owns" polymorphic="false">
+      <Class class="CatchBasin"/>
+    </Source>
+    <Target multiplicity="(1..1)" roleLabel="owned by" polymorphic="true">
+      <Class class="InletCaptureMethodAspect"/>
+    </Target>
+  </ECRelationshipClass>
+
+  <ECEntityClass typeName="PercentCapture" modifier="Sealed" displayLabel="Percent Capture">
+    <BaseClass>InletCaptureMethodAspect</BaseClass>
+    <ECProperty propertyName="CapturePercent" typeName="double" displayLabel="Capture Percent" description="Set the percent of incoming inflow to be captured by the inlet." category="HydraulicData"/>
```

Comment:
You could assign a kindOfQuantity to this unitized property. I notice neither the RoadRailUnits nor the AecUnits schemas have a KOQ for the involved "phenomenon" (PERCENTAGE). Thus, you could add one that you can run with - probably to RoadRailUnits. e.g.

```
<KindOfQuantity typeName="PERCENTAGE" displayLabel="Road &amp; Rail Percentage" persistenceUnit="u:PERCENT" presentationUnits="f:DefaultRealU(2)[u:PERCENT];f:DefaultRealU(4)[u:DECIMAL_PERCENT]" relativeError="0.0001"/>
```

Reply:
added the PERCENTAGE KindOfQuantity to the two fields....

also I was getting this error:

_***** Schema RoadRailUnits Validation Results *****
Schema file path: D:\learning\bis-schemas\Domains\2-DisciplinePhysical\Civil\RoadRailUnits\RoadRailUnits.ecschema.xml
"##vso[task.logissue type=error]Error BIS-1001: KindOfQuantity 'RoadRailUnits.PERCENTAGE' has persistence unit of unit system 'Units.INTERNATIONAL' but must have an SI unit system"_

so I updated the Units.ecschema.xml file to get passed the error